File: js/project4.js
/*
  A palindromic number reads the same both ways.
  The largest palindrome made from the product of
  two 2-digit numbers is 9009 = 91 × 99.
  
  Find the largest palindrome made from the
  product of two 3-digit numbers.
*/

function largestPalindrome() {
  let largestPalindromeProduct = 0
  for (i = 999; i > 0; i--){
    // for each number up to the highest 3 digit number 999
    for (j = 999; j > 0; j--) {
      // iterate through 1-999 multiplying together
      let product = i * j;
      // run palindrome test function on product
      if (isPalindrome(product) && product > largestPalindromeProduct){
        largestPalindromeProduct = product;
      }
    }
  }
  return largestPalindromeProduct;
}
// function to test whether a string is a palindrome and retrn true / false
function isPalindrome(test){
  str = test.toString();
  return str === str.split("").reverse().join("");
}

File: ruby/project4.rb
=begin
  A palindromic number reads the same both ways.
  The largest palindrome made from the product of
  two 2-digit numbers is 9009 = 91 × 99.

  Find the largest palindrome made from the
  product of two 3-digit numbers.
=end
# try to_s and to_i to prompt user for number of digits and iterate
def largestPalindrome()
  largestPalindromeProduct = 0
  i = 999
  loop do
    j = 999
    loop do
      product = i * j
      if isPalindrome(product) && product > largestPalindromeProduct
        largestPalindromeProduct = product
      end
      j -= 1
      break if j <=0
    end
    i -= 1
    break if i <= 0
  end
  return largestPalindromeProduct
end
# function to test whether a string is a palindrome and retrn true / false
def isPalindrome(test)
  str = test.to_s
  return str === str.reverse
end

puts largestPalindrome()
